src/lib/coda_base/coinbase.ml

```diff
@@ -15,6 +15,24 @@ module Stable = struct
 
     include T
     include Module_version.Registration.Make_latest_version (T)
+
+    let is_valid {proposer= _; amount; fee_transfer} =
+      match fee_transfer with
+      | None -> true
+      | Some (_, fee) -> Currency.Amount.(of_fee fee <= amount)
+
+    (* check validity when deserializing *)
+    include Binable.Of_binable
+              (T)
+              (struct
+                type nonrec t = t
+
+                let to_binable t = t
+
+                let of_binable t =
+                  assert (is_valid t) ;
```

Comment:
hmm I don't know if we want to assert here because an adversary can crash our client by sending malformed data here.

Reply:
I simply moved the existing code into `V1`.

Should we log and punish here instead?

Reply:
can you add an issue and a TODO pointing to that issue -- that should be good enough

Reply:
and leave the `assert` in for now?